src/webots/control/WbControlledWorld.cpp

```diff
@@ -220,6 +220,13 @@ void WbControlledWorld::addControllerConnection() {
       delete[] buffer;
     }
     foreach (WbRobot *const robot, robots()) {
+      if (robotName == robot->name() && robot->controllerName() == "<extern>" && robot->isControllerStarted()) {
+        updateRobotController(robot);
+        WbLog::info(tr("Restarting extern controller connection for robot \"%1\".").arg(robot->name()));
+        mRobotsWaitingExternController.append(robot);
+        startControllerFromSocket(robot, socket);
+        return;
+      }
       if (robot->isControllerStarted())
         continue;
```

Comment:
It is ok to duplicate the two commands, but I would at least  try to limit the duplication of the tests and add some comments:
```suggestion
      if (robot->isControllerStarted()) {
        if (robotName == robot->name() && robot->controllerName() == "<extern>") {
          // automatically disconnect from previous extern controller that may have failed
          updateRobotController(robot);
          WbLog::info(tr("Restarting extern controller connection for robot \"%1\".").arg(robot->name()));
          mRobotsWaitingExternController.append(robot);
          startControllerFromSocket(robot, socket);
        } // else ignore running controllers
        return;
      }
```

Reply:
Thanks a lot for your feedback :) ! I just saw your reply when I was thinking about extending this to the case with only 1 robot in the world. In this case, webots could know the controller to reboot.
I will integrate your change in the meantime.

Reply:
I think I've a nice solution regarding code duplication !
I'm not huge fan of the "else continue;", I would instead duplicate and add the "robot->isControllerStarted()" on line 233, especially because "updateRobotController" actually set this flag to false. But that not a big deal :)
It also integrate the special case with only 1 robot, so we don't need to specify name.
I've tested this with one robot, but not with multiples, however.

PS : I'm used to force-push new code when I've just "fixed" previous commit, in order to avoid bunch of fix commit. Is it fine for you ?
Moreover, I've mess up a bit here, hence the multiple "push", hope you don't get notification each time :/

PS2 : Test Suite Windows / build (windows-2019) (pull_request) don't seems to pass, but as far as I know, I'm not the only one. Should I fix something ?

Reply:
> PS : I'm used to force-push new code when I've just "fixed" previous commit, in order to avoid bunch of fix commit. Is it fine for you ?
Moreover, I've mess up a bit here, hence the multiple "push", hope you don't get notification each time :/

Given that the branch will be squashed and merged we don't care too much about the number of commits.

> PS2 : Test Suite Windows / build (windows-2019) (pull_request) don't seems to pass, but as far as I know, I'm not the only one. Should I fix something ?

The Windows tests is temporary broken due to some changes that we are not responsible for, so it is normal that it fails.